automation/jenkins_build.sh

```diff
@@ -75,8 +75,13 @@ else
 	done
 fi
 
-# if the build is marked as failed, print failed images
+# if the build is marked as failed, print failed images and tidy up everything
 if [ $exitCode -eq 1 ]; then
 	echo Failed images: ${failedList[@]}
+	# we need to clean up all untagged images when builds fail.
+	# First remove all stopped containers.
+	docker rm -v $(docker ps --filter status=exited -q)
```

Comment:
Can you grep this for containers that are using the base image, because there might be stopped containers unrelated to the base images and it'd be a bit antisocial to remove temporarily stopped images of other people